lib/ansible/modules/windows/win_auto_logon.ps1

```diff
@@ -9,75 +9,75 @@
 
 Set-StrictMode -Version 2.0
 
-$ErrorActionPreferrence = 'Stop'
-
-# argument spec hastable which has the input parameters and capabilities for this module
 $spec = @{
     options = @{
-        domain = @{type = "str"; default = $env:userdomain}
```

Comment:
Instead of having separate fields, make it easier for the user to specify the account in just `username`. I've added the code to correctly get the proper domain for that account and then splitting by `\` to get the domain part. The trouble with the `$env:userdomain` method is that it is reliant on the user who is executing the module not the user we want to set the auto logon as.